app/controllers/listings_controller.rb

```diff
@@ -932,9 +932,13 @@ def payment_setup_status(community:, user:, listing:, payment_type:, process:)
   end
 
   def delivery_config(require_shipping_address, pickup_enabled, shipping_price, shipping_price_additional, currency)
-    shipping = delivery_price_hash(:shipping, shipping_price, shipping_price_additional)
+    shipping = delivery_price_hash(:shipping, shipping_price, shipping_price_additional) if require_shipping_address
     pickup = delivery_price_hash(:pickup, Money.new(0, currency), Money.new(0, currency))
 
+    delivery_opts(require_shipping_address, pickup_enabled, shipping, pickup)
+  end
+
+  def delivery_opts(require_shipping_address, pickup_enabled, shipping, pickup)
```

Comment:
Actually, I think I just went through couple different alternatives before settling on that if clause on line 935 to guard against unnecessary calculation of shipping prices. 

During that process I refactored the case statement into a separate function and it just stuck.